packages/cliqz-home-react/src/Pagination.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component, PropTypes } from 'react';
+
+const propTypes = {
+  items: PropTypes.array.isRequired,
+  onChangePage: PropTypes.func.isRequired,
+  initialPage: PropTypes.number
+}
+
+const defaultProps = {
+  initialPage: 1
+}
+
+class Pagination extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pager: {}
+    }
+    this.timer = null;
+  }
+
+  componentWillMount() {
+    if (this.props.items && this.props.items.length) {
+      this.setPage(this.props.initialPage);
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.items !== prevProps.items) {
+      this.setPage(this.props.initialPage);
+    }
+  }
+
+  autoRotate() {
+    clearInterval(this.timer);
```

Comment:
clearInterval should probably be called also when component will be removed from dom. not sure what is the lifecycle hook for that in react